weblate/api/views.py

```diff
@@ -250,6 +252,111 @@ def repository(self, request, **kwargs):
         return Response(data)
 
 
+class UserViewSet(viewsets.ModelViewSet):
+    """Users API."""
+
+    queryset = User.objects.none()
+    serializer_class = UserSerializer
+    lookup_field = "username"
+
+    def get_queryset(self):
+        if self.request.user.has_perm("user.edit"):
+            return User.objects.all()
+        return self.request.user
+
+    def perm_check(self, request):
+        if not request.user.has_perm("user.edit"):
+            self.permission_denied(request, message="Can not manage Users")
+
+    def update(self, request, *args, **kwargs):
+        self.perm_check(request)
+        return super().update(request, *args, **kwargs)
+
+    def create(self, request, *args, **kwargs):
+        self.perm_check(request)
+        return super().create(request, *args, **kwargs)
+
+    def destroy(self, request, *args, **kwargs):
+        self.perm_check(request)
+        return super().destroy(request, *args, **kwargs)
```

Comment:
The user removal should be done by `weblate.accounts.utils.remove_user`. We do not actually delete the user, but remove all account data (to avoid need to rewrite lot of relations).

Not sure if we want self removal here,

Reply:
Fixed that.